Remove chef-provisioning from DK

tas50 opened the pull request:
Removes a good amount of Fog bloat, allows moving inspec/kitchen-ec2 over to aws-sdk-v3, and reduces overall DK maintenance burden. 

Signed-off-by: Tim Smith <tsmith@chef.io>

coderanger commented:
I think there is a case to be made for doing this out of the major rev cycle given the scope of impact. But if we go that route, we need to be very very vocal and clear about what's happening, like probably instead of removing the CLI command, have it display a banner explaining what's happened?

danielsdeleo commented:
I’m in favor as this was on my wishlist for v3 anyway. Agree with Noah on having a “tombstone” subcommand.

jtimberman commented:
Just now saw this and wanted to chime in - we (Chef Operations) are using chef-provisioning itself in a large number of older deployment pipelines for production, and we are installing it via ChefDK on the builders.

tas50 commented:
Rebased this. We've removed lazy loading of the Chef Provisioning resources from our current Chef 15 builds so we're ready to :shipit: here when we branch for the next workstation or DK.

robbkidd commented:
I'm thinking that a deprecation/removal notice should be added to the subcommand before its logic is removed and replaced with a tombstone.

tyler-ball commented:
Because Chef Provisioning will be EOL, we're not going to ship it in ChefDK 4 or Workstation 1.0 which will ship around April.

@robbkidd I'll open another PR that adds a deprecation warning to the command now
